scheduler/pkg/kafka/gateway/manager.go

```diff
@@ -49,13 +53,62 @@ type ConsumerConfig struct {
 	NumWorkers            int // infer workers
 }
 
-func NewConsumerManager(logger log.FieldLogger, consumerConfig *ConsumerConfig, maxNumConsumers int) *ConsumerManager {
-	return &ConsumerManager{
+func cloneKafkaConfigMap(m kafka.ConfigMap) kafka.ConfigMap {
+	m2 := make(kafka.ConfigMap)
+	for k, v := range m {
+		m2[k] = v
+	}
+	return m2
+}
+
+func NewConsumerManager(logger log.FieldLogger, managerConfig *ConsumerConfig, maxNumConsumers int) (*ConsumerManager, error) {
+	cm := &ConsumerManager{
 		logger:          logger.WithField("source", "ConsumerManager"),
-		consumerConfig:  consumerConfig,
+		managerConfig:   managerConfig,
 		consumers:       make(map[string]*InferKafkaHandler),
 		maxNumConsumers: maxNumConsumers,
 	}
+	err := cm.createKafkaConfigs(managerConfig)
+	if err != nil {
+		return nil, err
+	}
+	return cm, nil
+}
+
+func (cm *ConsumerManager) createKafkaConfigs(kafkaConfig *ConsumerConfig) error {
+	logger := cm.logger.WithField("func", "createKafkaConfigs")
+	var err error
+
+	producerConfigMap := config.CloneKafkaConfigMap(kafkaConfig.KafkaConfig.Producer)
+	producerConfigMap["go.delivery.reports"] = true
+	err = config.AddKafkaSSLOptions(producerConfigMap)
+	if err != nil {
+		return err
+	}
+
+	producerConfigAsJSON, err := json.Marshal(&producerConfigMap)
+	if err != nil {
```

Comment:
what happens when we fail serialisation?

Reply:
Its purely informational so the output will be less formatted.